internal/policy/verify.go

```diff
@@ -682,6 +682,15 @@ func verifyTagEntry(ctx context.Context, repo *git.Repository, policy *State, en
 		return err
 	}
 
+	entryTagRef, err := repo.Reference(plumbing.ReferenceName(entry.RefName), true)
+	if err != nil {
+		return err
+	}
+
+	if entry.TargetID != entryTagRef.Hash() && entry.TargetID != tagObj.Target {
+		return fmt.Errorf("verifying RSL entry failed, %s", ErrUnauthorizedSignature.Error())
```

Comment:
I'd not use ErrUnauthorizedSignature here, let's use an error that reflects the mismatch in metadata?

Reply:
Fixed